### PR TITLE
[#139841751] Add ADR for decision to redirect http app traffic.

### PR DESCRIPTION
**Pull request by alext**
## What

For discussion/review only at this stage:

Add an ADR describing the decision to redirect http app traffic to https.

### CODE REVIEW

**Comment by bleach** [comment]
👍  from me in terms of the intended behaviour.

Any idea about implementation? One place that already exists that could do it is [HAProxy](https://www.haproxy.com/doc/aloha/7.0/haproxy/http_redirection.html)

**Comment by bleach** [comment]
Should we close this and then re-open when we prioritise the work to do it?

File: docs/architecture_decision_records/ADR444-redirect-http-for-applications.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from ADR443.
+
+
+There are a number of things that could be done to address this:
+
+## Update the CF CLI to include the scheme
+
+Currently, the CF CLI outputs the fully-qualified hostname of the app after
+pushing, but doesn't include the scheme. This has caused confusion for users
+when this is copy/pasted into a browers, and then times out.
```
Comment (bandesz) [inline docs/architecture_decision_records/ADR444-redirect-http-for-applications.md]: "into browsers"

File: docs/architecture_decision_records/ADR444-redirect-http-for-applications.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+Currently, the CF CLI outputs the fully-qualified hostname of the app after
+pushing, but doesn't include the scheme. This has caused confusion for users
+when this is copy/pasted into a browers, and then times out. 
+
+Getting the CLI to include the scheme here will help with the specific case of
+users getting confused immediately after pushing an app.
+
+It's unclear how much work this involves, as currently information about
+whether a route is http or https doean't appear to be modeled in CloudFoundry
```
Comment (bandesz) [inline docs/architecture_decision_records/ADR444-redirect-http-for-applications.md]: "doesn't"

File: docs/architecture_decision_records/ADR444-redirect-http-for-applications.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+anywhere.
+
+If this involves changes to the CLI, there's no guarantee that users will
+upgrade their CLI.
+
+Additionally, there is some debate about how effective this change would be. It
+will probably fix some cases
```
Comment (bandesz) [inline docs/architecture_decision_records/ADR444-redirect-http-for-applications.md]: missing terminal point

File: docs/architecture_decision_records/ADR444-redirect-http-for-applications.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+## Redirect http to https
+
+Add an endpoint that listens to all http requests on cloudapps.digital and
+redirects them  to the corresponding https URL.
```
Comment (bandesz) [inline docs/architecture_decision_records/ADR444-redirect-http-for-applications.md]: extra space

File: docs/architecture_decision_records/ADR444-redirect-http-for-applications.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+# Decision
+
+We will redirect http traffic to the corresponding root https endpoint.
+
+We will continue to maintain HSTS preload lists for our production domains.
+
+# Status
+
+Pending
```
Comment (bandesz) [inline docs/architecture_decision_records/ADR444-redirect-http-for-applications.md]: This should be Accepted now probably.

**Comment by bandesz** [comment]
should I do these changes?

**Comment by alext** [comment]
I think I've addressed all your comments now.

**Comment by bandesz** [comment]
Also I added a fix:
```
-but this is only helps for users with recent versions of modern browsers.
+but this only helps users with recent versions of modern browsers.
```

**Pull request merged**